### PR TITLE
#2127,#1098 margin positions refactoring

### LINKED ISSUE #1098: Put bitCNY At The Top Of The List in Margin Positions
Since bitCNY has a much higher volume than bitBTC, I think it should be at the top of the Margin Positions list by default.

### LINKED ISSUE #2127: [10][dot5enko] Margin Position Updates
The UI Margin Position code is overdue for a refactoring to simplify the underlying code.

- [ ] Refactor app/components/Account/MarginPositions.jsx 

## Related tasks are 
- [ ] #2007 - No update on available DEBT asset
- [ ] #2004 - Redesign & Antify Borrow Modal
- [ ] #2005 - No working COLLATERAL lock

### PR DESCRIPTION
**Pull request by dot5enko**
- totally rewrote mechanism of getting margin positions printed which make it possible to sort them (closes #1098 - implemented sorting them by debt)
- reused a lot of similar code for representation
- removed some strange components  like wrappers for wrappers
- combined placeholder code and actual margin as they are the same except no data in placeholders
- made separate files for table (CollateralTable) and table row (MarginPosition)

Closes #2127 ((c) startail)

### CODE REVIEW

File: app/components/Account/CollateralTable.jsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    static propTypes = {
+        callOrders: ChainTypes.ChainObjectsList
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            assetsPropsCount: 0,
+            callOrdersCount: 0
```
Comment (sschiessl-bcp) [inline app/components/Account/CollateralTable.jsx]: this is unused?

File: app/components/Account/CollateralTable.jsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                ordersJson: callOrdersJson
+            };
+        }
+
+        return prevState;
+    }
+
+    render() {
+        let {account, callOrders} = this.props;
+        callOrders = callOrders.filter(o => !!o);
```
Comment (sschiessl-bcp) [inline app/components/Account/CollateralTable.jsx]: Does this really happen, I mean that `o` does not exist?
In `getDerivedStateFromProps` this check is not included. 

In any case: 
- Is this used anywhere further after filtering?
- If so, would this not be more suitable to be filtered when receiving props, or when actually using it to render?

File: app/components/Account/CollateralTable.jsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+        let rows = this.state.assets
+            .sort((a, b) => {
+                // a,b - order ChainObject's
+                // could be done via component's property `order_by`
+
+                let aVal = a.has_margin_order ? a.order.get("debt") : 0;
+                let bVal = b.has_margin_order ? b.order.get("debt") : 0;
+
+                return bVal - aVal;
```
Comment (sschiessl-bcp) [inline app/components/Account/CollateralTable.jsx]: if both have no order, fallback to sort by symbol

File: app/components/Account/AccountOverview.jsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import Immutable from "immutable";
 import Translate from "react-translate-component";
 import TotalBalanceValue from "../Utility/TotalBalanceValue";
 import AssetName from "../Utility/AssetName";
-import MarginPositions from "./MarginPositions";
+import CollateralTable from "./CollateralTable";
```
Comment (sschiessl-bcp) [inline app/components/Account/AccountOverview.jsx]: I'd suggest to be consistent in the naming. Can this be MarginPositionTable?

Comment (dot5enko) [reply app/components/Account/AccountOverview.jsx]: sure

File: app/components/Account/CollateralTable.jsx (hunk 1 above)
Comment (dot5enko) [reply app/components/Account/CollateralTable.jsx]: why not? used
at the bottom of the exported component.

Comment (sschiessl-bcp) [reply app/components/Account/CollateralTable.jsx]: Can't find it, maybe not commited?

image

File: app/components/Account/CollateralTable.jsx (hunk 2 above)
Comment (dot5enko) [reply app/components/Account/CollateralTable.jsx]: yup, sometimes (when ws client just connect to api) we got some "undefined" results when using.
yes, getDerivedStateFromProps should use this too

File: app/components/Account/CollateralTable.jsx (hunk 3 above)
Comment (dot5enko) [reply app/components/Account/CollateralTable.jsx]: ok

File: app/components/Account/CollateralTable.jsx (hunk 2 above)
Comment (dot5enko) [reply app/components/Account/CollateralTable.jsx]: and yes, `callOrders` not used in render method anymore -  handle those data in `getDerivedStateFromProps`, so this line is going to be removed

Comment (sschiessl-bcp) [reply app/components/Account/CollateralTable.jsx]: Where is callOrders used again in the rendering? Can't find it in MarginPosition

File: app/components/Account/CollateralTable.jsx (hunk 3 above)
Comment (dot5enko) [reply app/components/Account/CollateralTable.jsx]: hm, then i'll add 0 before symbol in order to keep order of rows with debt, @sschiessl-bcp what do you think?

File: app/components/Account/CollateralTable.jsx (hunk 1 above)
Comment (dot5enko) [reply app/components/Account/CollateralTable.jsx]: i meant "yes" - unused. removed

File: app/components/Account/CollateralTable.jsx (hunk 3 above)
Comment (sschiessl-bcp) [reply app/components/Account/CollateralTable.jsx]: Sorry for being vague.

Sorting:
 - having an order goes above having no order
 - if both have an order, sort by debt
 - if both have no order, sort by symbol

**Comment by dot5enko** [comment]
package and lock changes were removed

**Comment by startailcoon** [comment]
@sschiessl-bcp are we ready to merge this PR?

**Comment by dot5enko** [comment]
@sschiessl-bcp ? do you intend to review it ?

**Comment by sschiessl-bcp** [comment]
Yes, release candidate integration is prioritized right now. I ll get back to the prs after

**Pull request merged**